src/reconcile/rec_col.c

```diff
@@ -789,8 +809,30 @@ __wt_rec_col_var(
                     size = upd->size;
                     break;
                 case WT_UPDATE_TOMBSTONE:
-                    twp = &clear_tw;
+                    /*
+                     * When removing a key due to a tombstone with a durable timestamp of "none",
+                     * also remove the history store contents associated with that key.
+                     */
+                    if (twp->durable_stop_ts == WT_TS_NONE && hs_clear) {
```

Comment:
I don't quite understand why this only applies to timestamp none

Reply:
I don't know, but it's the same logic as in the row code. I would naively expect the _start_ timestamp to be the relevant part. I guess if the stop time isn't none there's at least one other more recent value, and if there's more than one all but the latest live in the history store and shouldn't be purged, and probably it's not easy to tell here if there's more than one?

Reply:
OK, then the row store logic must have changed since last time I checked. @hbokhari Do you know why?

Reply:
The code in the row store is refactored, but the logic is the same. We never checked for transaction id.